app/src/store/cart.ts

```diff
@@ -95,6 +96,102 @@ export default function useCartStore() {
     }
   }
 
+  // convert the amount to a different token (direction allows for moving to/from the tokenAddress)
+  function getConvertedAmount(amount: number, tokenAddress: string, direction = 1) {
+    const exchangeRate = quotes.value[tokenAddress] ?? 0;
+
+    return amount * (direction == 1 ? exchangeRate : 1 / exchangeRate);
+  }
+
+  // Linear Interpolation function to find the point on the curve that amount corresponds to
+  function lerp(x_lower: number, x_upper: number, y_lower: number, y_upper: number, x: number) {
+    return y_lower + ((y_upper - y_lower) * (x - x_lower)) / (x_upper - x_lower);
+  }
+
+  // returns the prediction curve for this grant in the given round
+  function getPredictionsForGrantInRound(grantId: string, round: GrantRound) {
```

Comment:
For all the methods with `grantId` as an input, I'd suggest changing the type to `BigNumberish` and converting the input to the required type with `BigNumber.from(grantId).toString()? A bit more robust that way / removes the burden of handling types from the caller